src/components/Draggable/RealFilterDraggable.tsx

```diff
@@ -0,0 +1,205 @@
+    setSearchedFilterSet(filterValues)
+  }
+
+  const handleSelect = (element: string) => (event: React.ChangeEvent<HTMLInputElement>) => {
+    filterState.has(element) ? filterState.delete(element) : filterState.add(element)
+    handleFilterUpdate(name, new Set<string>(filterState))
+    setAll(
+      filterState.size === 0
+        ? QuantityEnum.EMPTY
+        : filterState.size === filterValues.length
+        ? QuantityEnum.FULL
+        : QuantityEnum.HALF_FULL
+    )
+  }
+
+  const handleKeyDown = (filterKey: keyof T) => (event: any) => {
+    onKeyNav(getKeyDirection(event.nativeEvent.key), origin, filterKey)
+    onDragEnd()
+  }
+
+  const handleSearch = () => (event: any) => {
+    const searchResults = new Array<string>()
+    const searchText: string = (event.currentTarget.value as string).toLocaleLowerCase()
+    filterValues.forEach((element: string) => {
+      const stringElement = element + ''
+      const loweredElement = stringElement.toLocaleLowerCase()
+      const found = loweredElement.search(searchText) !== -1
+      found && searchResults.push(element)
+    })
+    setSearchedFilterSet(searchResults)
+  }
+
+  const handleSelectAll = () => () => {
+    if (all === 2) {
+      setAll(QuantityEnum.EMPTY)
+      handleFilterUpdate(name, new Set<string>(new Set<string>()))
+    } else {
+      setAll(QuantityEnum.FULL)
+      handleFilterUpdate(name, new Set<string>(filterValues))
+    }
+  }
+
+  return (
+    <div ref={drag} className={css(classes.dndBox, isDragging && classes.dndBoxDragging)}>
+      <React.Fragment>
+        <Button
+          style={classes.button}
+          innerRef={setButtonRef}
+          onClick={handleClick}
+          onKeyDown={handleKeyDown(name)}
+          size='small'
+          kind='primary'
+          skin='ghost'
+        >
+          <HFlow hSpacing={0.5}>
+            <Icon icon='dragdrop' />
+            {value}
+            {open ? <Icon icon='angleUp' /> : <Icon icon='angleDown' />}
+          </HFlow>
+        </Button>
+        <Dropdown
+          anchorRef={buttonRef}
+          open={open}
+          autoclose={false}
+          onClose={handleClose}
+          popperProps={{ placement: 'bottom' }}
+          style={classes.dropdown}
+        >
+          <div title='dropDownArea' className={classes.dropdownArea} onBlur={(e) => e.stopPropagation()}>
+            <DropdownItem className={classes.noOutline}>
+              <div className={classes.search}>
+                <TextField
+                  name='iconized'
+                  id='iconized'
+                  placeholder='Pesquisa'
+                  icon='zoomOutline'
+                  onChange={handleSearch()}
+                />
+              </div>
+            </DropdownItem>
+            <FixedSizeList
+              height={
+                filterValues.length + 1 > 5
+                  ? theme.typography.sizes.html * 8
+                  : theme.typography.sizes.html * 2.2 * (filterValues.length + 1)
```

Comment:
:mage:

Reply:
I talked to Batista, who developed the component, and he said that these numbers (bigValue, FixedSlizeList height and itemSize) were chosen arbitrarily.